jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/TargetDirectory.java

```diff
@@ -153,10 +161,23 @@ static boolean deleteMyFiles() {
 	static void deleteMyDirectory(boolean directoryEmpty) {
 		if (LOGGING_DISABLED != loggingStatus) {
 			IPC.logMessage("deleting my directory "); //$NON-NLS-1$
+		}
+		/* try the fast path for deleting files and directories. try the heavyweight path if something fails */
+		if ((!directoryEmpty && !deleteMyFiles()) || (null == targetDirectoryFileObject) || !targetDirectoryFileObject.delete()) {
+			String vmId = AttachHandler.getVmId();
+			IPC.logMessage("fast path for deleting files and dirs fail, trying the heavyweight path for vmId: ", vmId);
+			deleteTargetDirectory(vmId);
+		}
+	}
+
+	private static void deleteMyDirectory(boolean directoryEmpty, String vmId) {
```

Comment:
I think `deleteMyDirectory(boolean directoryEmpty)` can be removed, and replaced with `deleteMyDirectory(boolean directoryEmpty, "")`. When `vmId.isEmpty()`, apply `vmId = AttachHandler.getVmId()`.

Reply:
I agree.